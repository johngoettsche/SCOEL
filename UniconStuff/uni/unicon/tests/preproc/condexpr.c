/* test the various expressions allowed in conditions */
#if 1
yes
#else
no
#endif

#if 0
yes
#else
no
#endif

#if '0'
yes
#else
no
#endif

#if '\0'
yes
#else
no
#endif

#if x
yes
#else
no
#endif

#define x 1

#if x
yes
#else
no
#endif

#if '\xf' == 15
yes
#else
no
#endif

#if '\xff' == 255
yes
#else
no
#endif

#if '\xa' != '\12'
yes
#else
no
#endif

#if 1 != y
yes
#else
no
#endif

#if 3 < 10
yes
#else
no
#endif

#if 123456789 < 123456788
yes
#else
no
#endif

#if '\x7e' <= '\176'
yes
#else
no
#endif

#if 4 <= 3
yes
#else
no
#endif

#if     10    >    9     
yes
#else
no
#endif

#if		8		>		9		
yes
#else
no
#endif

#if 10 >= 9
yes
#else
no
#endif

#if 8 >= 9
yes
#else
no
#endif

#if defined y == 0
yes
#else
no
#endif

#define y(a) 0

#if defined y == 0
yes
#else
no
#endif

#if defined(y) == 1
yes
#else
no
#endif

#if 0x4e85  == 047205
yes
#else
no
#endif

#if 0xabcde == 0XABCDE
yes
#else
no
#endif

#if 1 > 2 ? 0 : 1
yes
#else
no
#endif

#if 1 < 2 ? 0 : 1
yes
#else
no
#endif

#if 0 || 1
yes
#else
no
#endif

#if 0 || 0
yes
#else
no
#endif

#if 1 && 1
yes
#else
no
#endif

#if 1 && 0
yes
#else
no
#endif

#if (9 | 3) == 11
yes
#else
no
#endif

#if (0720 & 0230) == 0220
yes
#else
no
#endif

#if (0x99 ^ 0x66) == 0xff
yes
#else
no
#endif

#if (3 << 3) == 24
yes
#else
no
#endif

#if (25 >> 3) == 3
yes
#else
no
#endif

#if (300 - 7) == 293
yes
#else
no
#endif

#if ('0' + 3) == '3'
yes
#else
no
#endif

#if (3 * 4) == 12
yes
#else
no
#endif

#if(77/11)==7
yes
#else
no
#endif

#if (29 % 9) == 2
yes
#else
no
#endif

#if +6 == 6
yes
#else
no
#endif

#if (15 - 26) == -11
yes
#else
no
#endif

#if (~5 & 0xf) == 10
yes
#else
no
#endif

#if !1
yes
#else
no
#endif

#if !0
yes
#else
no
#endif

#if (901 == 901L) && (7u == 7l) && (12U == 12Lu) && (5Ul == 5)
yes
#else
no
#endif

#define average(a,b) (((a) + (b))/2)
#define abs(a) (a) ? (a) : -(a)

#if abs(average(2 - 100, -12)) == average(abs(100 - 2), abs(-12))
yes
#else
no
#endif
