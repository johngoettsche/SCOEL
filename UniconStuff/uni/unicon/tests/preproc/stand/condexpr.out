#line 2 "condexpr.c"

yes

#line 10 "condexpr.c"

no



yes

#line 22 "condexpr.c"

no

#line 28 "condexpr.c"

no

#line 34 "condexpr.c"

yes

#line 40 "condexpr.c"

yes

#line 48 "condexpr.c"

no

#line 54 "condexpr.c"

no



yes

#line 64 "condexpr.c"

yes

#line 72 "condexpr.c"

no



yes

#line 84 "condexpr.c"

no



yes

#line 96 "condexpr.c"

no



yes

#line 108 "condexpr.c"

no



yes

#line 122 "condexpr.c"

no



yes

#line 132 "condexpr.c"

yes

#line 138 "condexpr.c"

yes

#line 144 "condexpr.c"

yes

#line 152 "condexpr.c"

no



yes

#line 164 "condexpr.c"

no



yes

#line 176 "condexpr.c"

no



yes

#line 186 "condexpr.c"

yes

#line 192 "condexpr.c"

yes

#line 198 "condexpr.c"

yes

#line 204 "condexpr.c"

yes

#line 210 "condexpr.c"

yes

#line 216 "condexpr.c"

yes

#line 222 "condexpr.c"

yes

#line 228 "condexpr.c"

yes

#line 234 "condexpr.c"

yes

#line 240 "condexpr.c"

yes

#line 246 "condexpr.c"

yes

#line 252 "condexpr.c"

yes

#line 260 "condexpr.c"

no



yes

#line 270 "condexpr.c"

yes

#line 279 "condexpr.c"

yes
